### PR TITLE
For #3722: Invoke search on using long press home button for assist app  

### PR DESCRIPTION
**Pull request by s-ankur**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by gabrielluong** [comment]
@s-ankur is this still a draft?

**Comment by s-ankur** [comment]
@gabrielluong yes, pls wait a bit. I didn't get the time to finish it due to work 😅

**Comment by s-ankur** [comment]
Hi @gabrielluong could you pls review

File: app/src/main/java/org/mozilla/fenix/components/metrics/MetricsUtils.kt [hunk 1]
```diff
@@ -57,6 +57,11 @@ object MetricsUtils {
                     engineSource
                 )
             )
+            SearchAccessPoint.ASSIST -> Event.PerformedSearch(
+                Event.PerformedSearch.EventSource.Other(
+                    engineSource
```
Comment (s-ankur) [inline app/src/main/java/org/mozilla/fenix/components/metrics/MetricsUtils.kt:62]: Is Other" ok or should another Event Source be created?

File: app/src/main/java/org/mozilla/fenix/home/intent/StartSearchIntentProcessor.kt [hunk 2]
```diff
@@ -15,49 +15,49 @@ import org.mozilla.fenix.ext.nav
...
-                    source = Event.PerformedSearch.SearchAccessPoint.SHORTCUT
+                    Event.PerformedSearch.SearchAccessPoint.SHORTCUT
                 }
                 PRIVATE_BROWSING_PINNED_SHORTCUT -> {
                     metrics.track(Event.PrivateBrowsingPinnedShortcutPrivateTab)
-                    source = Event.PerformedSearch.SearchAccessPoint.SHORTCUT
+                    Event.PerformedSearch.SearchAccessPoint.SHORTCUT
+                }
+                else -> {
+                    if (intent.action == Intent.ACTION_ASSIST)
```
Comment (s-ankur) [inline app/src/main/java/org/mozilla/fenix/home/intent/StartSearchIntentProcessor.kt:44]: Should this entry point be tracked as well?

**Comment by s-ankur** [comment]
Hi @jonalmeida, attaching the gif. Please do tell me if I need to do anything else to speed up review
GIF-201130_165949

**Comment by s-ankur** [comment]
Fixed merge issue and added test

**Comment by boek** [comment]
Rebased and refactored into #28101

**Pull request closed without merging**